### DISCUSSION TITLE
[Enhancement] Chat UI

### DISCUSSION THREAD

**Discussion by coen22** [dotnet/maui#1519] [Ideas]
## Summary
In Xamarin Forms it's currently quite difficult to make a smooth Chat UI without a custom renderer (at least this has been my experience). And even then it had performance issues on Android.

## API Changes

Could you maybe create an API that wraps around some native Chat UI components on both iOS and Android?
https://github.com/stfalcon-studio/ChatKit (Android)
https://github.com/MessageKit/MessageKit (iOS)

e.g. you could have something like this.

    var chat = new MessagesList {
        ShowDateHeader = true,
        Bubbles {
            new ChatBubble {
                SendAt = DateTime.UtcNow,
                Content = new Label {
                    Text = "Hey"
                }
            }
        }
    }

    Content = chat;

## Intended Use Case
I think many applications can benefit from a user-friendly Chat UI.

**Comment by matt-goldman** [comment]
This might be better suited to the community toolkit

**Reply by mcguu** [reply]
I hate `community toolkit`

**Comment by coen22** [comment]
Thanks, I'll post the request there :)
